enhancements/microshift/microshift-supportability-tools.md

```diff
@@ -0,0 +1,384 @@
+1. Upgrades and incompatible changes.
+1. `sos` release cadence.
+1. Extra required disk space.
+
+#### Upgrades
+When upgrading a MicroShift system we may need to upgrade just MicroShift, or
+the OS as well. If it is the latter then `sos` will be updated before
+MicroShift, therefore we must ensure backwards compatibility with the plugin.
+
+If it is the former then we need to ensure newer MicroShift versions do not
+introduce incompatible changes with the reporting capabilities of the `sos`
+plugin. This may also be seen from a different perspective: the plugin should
+make no assumptions about names or locations of components whenever possible.
+In other words, the plugin must be written in the most generic way that still
+allows retrieval of relevant information.
+
+Still, if incompatible changes are introduced then we need to upgrade `sos` as
+well.
+
+#### Sos release cadence
+`sos` release cadence has been shortened to 4 weeks starting on Feb 1st 2023.
+This change has been introduced to allow faster support of new components and
+features. Since MicroShift follows the same release calendar than OpenShift,
+`sos` is faster.
+
+New versions of `sos` will simply slot in to the next regular release. If there
+is an urgent need a hotfix can be made between releases.
+
+#### Extra disk space
+Adding a plugin to `sos` and not relying on MicroShift binary means it needs
+the `oc` or `kubectl` binary installed. This package is available in the RHEL
+repos, but it comes at an extra cost of 125MB once installed.
+
+If we compare it against the minimum disk size required (which is 8GB at the
+time of this writing), it represents an extra 1.5%.
+
+### Drawbacks
+Discussed in risks and mitigations.
+
+## Design Details
+Refer to `sos` [documentation](https://github.com/sosreport/sos/wiki/How-to-Write-a-Plugin)
+for how to create plugins.
+
+The file structure should ressemble that of must-gather.
+
+### Open Questions [optional]
+N/A
+
+### Test Plan
+MicroShift only tests should be added to verify the output structure: dirs and
+files. The contents may be checked only at the file size level.
+
+An e2e test should test the command exits successfully and certain contents are
+always present.
+
+Once the plugin is introduced, it will also be used as part of the debug logs
+for all failed jobs in the MicroShift CI.
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+- Have the plugin as part of `sos` released in RHEL9 repos.
+- QE and users feedback.
+- Initial test coverage.
+
+#### Tech Preview -> GA
+
+- User facing documentation.
+- Introduce for all other CI jobs.
+- Complete test coverage.
+
+#### Removing a deprecated feature
+N/A
+
+### Upgrade / Downgrade Strategy
+This is handled at the OS level, as `sos` is a system package.
+
+### Version Skew Strategy
+`sos` package comes with RHEL OS. We might see two different upgrade scenarios:
+1. MicroShift z-stream where the OS does not change.
+1. Any other upgrade involving the OS.
```

Comment:
I do not understand these comments. Isn't `sos` delivered as an RPM? We already have sos dependency in MicroShift RPM, so we can add a specific version requirement.

Reply:
It was badly written, rephrased.